stonesoup/metricgenerator/manager.py

```diff
@@ -46,8 +60,15 @@ def add_data(self, groundtruth_paths: Iterable[Union[GroundTruthPath, Platform]]
             overwriting one field (e.g. tracks) does not affect the others
         """
 
-        self._add(overwrite, groundtruth_paths=groundtruth_paths,
-                  tracks=tracks, detections=detections)
+        # Clear all generator caches
+        for generator in self.generators:
+            try:
+                generator.clear_caches()
+            except AttributeError:
+                pass
+
+        self._add(overwrite, _groundtruth_paths=groundtruth_paths,
```

Comment:
When `overwrite` is `True`, the `_add` method should probably copy the set, otherwise you are storing mutable version which may be modified elsewhere.

Reply:
Does this need to be a deep copy? Individual ground truth paths / tracks / detections could be modified?

Reply:
I think a shallow copy would be okay.